File: common/src/main/java/dev/niostone/interstellar/config/builder/StarSystemConfigBuilder.java
package dev.niostone.interstellar.config.builder;

import dev.niostone.interstellar.config.StarSystemConfig;
import net.minecraft.resources.ResourceLocation;

public class StarSystemConfigBuilder {
    private StarSystemConfig config;

    public StarSystemConfigBuilder(String name) {
        this.config = new StarSystemConfig(name);
    }

    public StarSystemConfigBuilder(ResourceLocation name) {
        this.config = new StarSystemConfig(name.toString());
    }

    public StarSystemConfigBuilder(String namespace, String id) {
        this.config = new StarSystemConfig(namespace + ":" + id);
    }

    public StarSystemConfigBuilder id(int id) {
        this.config.id = id;

        return this;
    }

    public StarSystemConfigBuilder planet(int planetId) {
        this.config.planets.add(planetId);

        return this;
    }

    public StarSystemConfig build() {
        return this.config;
    }
}

File: common/src/main/kotlin/org/stardustmodding/interstellar/api/builder/StarSystemConfigBuilder.kt
package org.stardustmodding.interstellar.api.builder

import net.minecraft.util.Identifier
import org.stardustmodding.interstellar.impl.config.StarSystemConfig

class StarSystemConfigBuilder {
    val config: StarSystemConfig

    constructor(name: String) {
        this.config = StarSystemConfig(name)
    }

    constructor(name: Identifier) {
        this.config = StarSystemConfig(name.toString())
    }

    constructor(namespace: String, id: String) {
        this.config = StarSystemConfig("$namespace:$id")
    }

    fun planet(value: Int) = apply { this.config.planets.add(value) }

    fun build(): StarSystemConfig {
        return this.config
    }
}
